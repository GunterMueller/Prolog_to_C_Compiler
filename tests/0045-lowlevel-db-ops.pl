main :-
	foreign_call(db_create('foo', 3001, DB)),
	foreign_call(db_record(DB, 1, 'bar', 123, _)),
	foreign_call(db_record(DB, 1, 'bar', this(is, a, "term"), _)),
	foreign_call(db_record(DB, 1, 'baz', [X,Y,X], _)),
	foreign_call(db_find(DB, 'bar', REF)),
	foreign_call(db_ref(REF, X)),
	display(X), nl,
	foreign_call(db_next(REF, REF2)),
	foreign_call(db_ref(REF2, Y)),
	display(Y), nl,
	\+foreign_call(db_next(REF2, _)),
	foreign_call(db_find(DB, 'baz', REF3)),
	foreign_call(db_ref(REF3, Z)),
	display(Z), nl,
	\+foreign_call(db_find(DB, yo, _)).
