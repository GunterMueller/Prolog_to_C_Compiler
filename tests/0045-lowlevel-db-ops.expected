123
this(is, a, [116,101,114,109])
[_13,_14,_13]
